src/compiler/checker.ts

```diff
@@ -30345,7 +30345,14 @@ namespace ts {
         }
 
         function assignContextualParameterTypes(signature: Signature, context: Signature) {
-            signature.typeParameters = context.typeParameters;
+            if (context.typeParameters) {

```

Comment:
Is context.typeParameters ever `undefined`? Was that the bug here?

Reply:
I mean, I don't want to overwrite a signature's type parameters with `undefined`, and there's no reason to do anything if they're _both_ `undefined`.